docs/config/README.md

```diff
@@ -44,6 +44,29 @@ Or for Cmd (Windows) would be adding this line to your `starship.lua`:
 os.setenv('STARSHIP_CONFIG', 'C:\\Users\\user\\example\\non\\default\\path\\starship.toml')
 ```
 
+### Loading additional configuration from the environment
+
+In addition to the configuration file, starship also allows you to set configuration values by using environment variables. The variables should be formatted like this: `STARSHIP_CONFIG__SECTION_NAME__KEY=value`. The configuration keys in the environment need to start with `STARSHIP_CONFIG__` and table keys need to be separated with `__`.
+
+In most POSIX-like shells, you can set the `disabled` key of the `status` module to `false` with the following:
+
+```sh
+export STARSHIP_CONFIG__STATUS__DISABLED=false
+```
+
+Equivalently in PowerShell (Windows) would be adding this line:
+
+```powershell
+$ENV:STARSHIP_CONFIG__STATUS__DISABLED = "false"
+```
+
+Other shells may use a different syntax to set environment variables.
+Any values set in the environment will be parsed as TOML-values, not as full documents. If the environment values cannot be directly parsed as TOML-values, they are treated as strings.
+
+This means that `true` is parsed as a boolean when it's not quoted, but is parsed as a string when it is quoted. In contrast, when a value that cannot be parsed as a TOML value, such as `starship` is found, it is treated as a string (`"starship"`).
+
+Because the values are TOML-values the `[table]` syntax will not work, but inline tables `{ a = 1 }` or arrays `[ 0, 1 ]`.
```

Comment:
Could this eager parsing logic (and the expansiveness of the environment configurability) be a security risk?

If this PR were merged then an attacker with write access to environment variables would be able to escalate that access into remote code execution (RCE) by setting `STARSHIP_CONFIG__CUSTOM__MY_RCE_MODULE__WHEN=<...open_rce_shell_command>`.

Having write access to environment variables seems like pretty high privilege to me already so being able to escalate that into RCE may not be an issue, but I'm not sure. 🤷

Reply:
Some food for thought: https://www.oreilly.com/library/view/secure-programming-cookbook/0596003943/ch01s01.html